Squish username before validation

aurelien-reeves opened the pull request:
Fix #10101

Commit:
Squish username before validation (#10101)

Fix #10101

Gargron commented on app/models/account.rb:
```diff
@@ -78,6 +78,10 @@ class Account < ApplicationRecord
   validates :note, note_length: { maximum: 160 }, if: -> { local? && will_save_change_to_note? }
   validates :fields, length: { maximum: 4 }, if: -> { local? && will_save_change_to_fields? }
 
+  before_validation(on: :create) do
```
Style-wise, it would be preferable to do it via a private method

`before_validation :prepare_username, on: :create`

I am also not sure what `squish` gives here over `strip`, would prefer to just use `strip`

aurelien-reeves replied:
Indeed, I will move `before_validation` into a private method.

Regarding `squish`, actually it processes unicode whitespaces. `strip` does not.

```ruby
pry(main)> "\u3000test\u00a0".strip
=> " test "
pry(main)> "\u3000test\u00a0".squish
=> "test"
```

If you agree to use `squish` to process Unicode white-spaces, I will add a dedicated test for that, or at least I will add Unicode white-spaces into the test I have added.

Commit:
Move before_validation hook to a private method

Also add Unicode wite-spaces to the spec to support the use of squish
over strip.
